src/DependencyInjection/Compiler/SonataAddBlockDialogPass.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the ekino/sonata project.
+ *
+ * (c) Ekino
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\HelpersBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+/**
+ * Class SonataAddBlockDialogPass.
+ *
+ * @author Christian Kollross <christian.kollross@ekino.com>
+ */
+final class SonataAddBlockDialogPass implements CompilerPassInterface
+{
+    /**
+     * @param ContainerBuilder $container
+     */
+    public function process(ContainerBuilder $container)
+    {
+        $configs = $container->getExtensionConfig('sonata_helpers');
+        $enabled = $configs[0]['compose_container']['enabled'];
+
+        if ($enabled && $container->hasParameter('sonata.page.admin.page.controller')) {
```

Comment:
Why doing that here, in a compiler pass? Why don't do it in the extension?

Reply:
We only want to replace the `PageAdminController` if the feature is enabled. Is there another way for that?

Reply:
In the extension

Reply:
Cool, that works, and it saves us two classes.